### PR TITLE
Add upgrade_image option

### PR DESCRIPTION
**Pull request by varad-ahirwadkar**
For upgrading the cluster with the specific build, added `upgrade_image` variable.
Signed-off-by: Varad Ahirwadkar <varad.ahirwadkar@ibm.com>

### CODE REVIEW

**Comment by yussufsh** [comment]
@varad-ahirwadkar could you please mention in the doc that one of image or version is needed for upgrade? Also which one takes precedent. I think image will be considered when both are given. Would like confirmation on these tests from your side. Thanks.

**Comment by varad-ahirwadkar** [comment]
> @varad-ahirwadkar could you please mention in the doc that one of image or version is needed for upgrade? Also which one takes precedent. I think image will be considered when both are given. Would like confirmation on these tests from your side. Thanks.

ok sure, yes image will take precedent.

**Comment by yussufsh** [comment]
@varad-ahirwadkar you need to fix the format errors shown above. Sorry for being so picky about it.

**Comment by varad-ahirwadkar** [comment]
> @varad-ahirwadkar you need to fix the format errors shown above. Sorry for being so picky about it.

Yes, resolved format errors!

**Comment by varad-ahirwadkar** [comment]
I have made the changes in documentation. Could you please review PR @yussufsh @sudeeshjohn ?
Thanks.

File: modules/5_install/install.tf
Comment (yussufsh) [inline modules/5_install/install.tf:404]: @varad-ahirwadkar Did you test all the combinations here? As per my knowledge, this will always evaluate as `true` (hence count as 1).

Comment (varad-ahirwadkar) [reply modules/5_install/install.tf:404]: I tested with three scenarios:
By using blank values -> successfully deployed cluster
By setting both variables to check precedence -> successfully deployed cluster and upgraded to `upgrade_image`
By setting only upgrade_image ->  successfully deployed cluster and upgraded to `upgrade_image`

Comment (yussufsh) [reply modules/5_install/install.tf:404]: See this from Terraform console:
```
> ("" != "" || "" != "")
false
> ("" != "" || "" != "") != ""
true
```

You are comparing `false != ""` which is always `true`.

Comment (yussufsh) [reply modules/5_install/install.tf:404]: > By using blank values -> successfully deployed cluster

Deployment may be successful but we don't need to run the upgrade resource.

```suggestion
  count      = var.upgrade_version != "" || var.upgrade_image != "" ? 1 : 0
```

Comment (varad-ahirwadkar) [reply modules/5_install/install.tf]: Hi @yussufsh, I have made this change and re-tested the above three scenarios.
Thanks.

**Review by yussufsh — DISMISSED** [review]
/lgtm
/cc @cs-zhang

File: modules/5_install/install.tf
Comment (Prajyot-Parab) [inline modules/5_install/install.tf:406]: Don't we need to trigger this module on image change as well? @varad-ahirwadkar

Comment (varad-ahirwadkar) [reply modules/5_install/install.tf:406]: Hi @Prajyot-Parab, I have added `var.upgrade_version`  to trigger.

**Review by cs-zhang — APPROVED** [review]
/lgtm

**Review by yussufsh — APPROVED** [review]

**Comment by yussufsh** [comment]
@mkumatag another one waiting to be merged using the bot.

**Comment by mkumatag** [comment]
Per my initial investigation what I see is that prow does optimisation while merging and feature called batch merge `Automatically runs batch tests and merges multiple PRs together whenever possible.` and this PR is now batched with #412   because per prow condition these PRs got the `/lgtm` and `/approval` and ready to merge and clubbed into one batch for the final test and merge.

But with our condition in the GH flow where we aren't allowing the PR to merge until we get 2 approval(which is a trouble maker here)

In future we need to find someway to workaround this or remove this +2 concept because prow is working properly.

**Pull request merged**
